### DISCUSSION TITLE
frequency shift plot

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#183] [Ideas]
Jasper,

Do I understand correctly that the frequency shift plot shows the average? If yes, whether it is also possible to add a median value? 
Thank you!

MaxPi

**Comment by jvde-github** [comment]
Yes, it is the average but a bit challenging to add a median as I do not store the ppm numbers to keep the program as light as possible.

Perhaps this can be easily done in a Grafana dashboard, e.g. https://kx1t.com/ais/?tab=about

**Comment by Taxom** [comment]
Jasper,

Can be this added as an add-on feature which can be turned on in config? And store data on the frequency shift in RAM. Because Grafana may be overwhelming and requires too much i/o operation on SD-card.
Thanks.

MaxPi.

**Reply by jvde-github** [reply]
Hi, I would run Grafana and Prometheus on a different system. There is no simple way to add this calculation as a plugin as it needs to be implemented in the main C++ files. It can be a switch though. Essentially add a database that stores all samples and periodically calculates median (by sorting the database). The program does currently not have such a database and functionality. It is a hobby project so likely do not have the time for this.

Would be nice if someone can write a nice AIS analysis tool....
